src/utils/blockchain/refresh-data.ts

```diff
@@ -0,0 +1,57 @@
+import { Web3Interface } from "../../scripts/web3-interface";
+import { unwrapSarcophagus } from "./unwrap";
+import { scheduleUnwrap } from "../scheduler";
+import { getGracePeriod, getSarcophagiIds, inMemoryStore, SarcophagusData, SarcophagusState } from "../onchain-data";
+import { BigNumber } from "ethers";
+
+// TODO -- once typechain defs are in the sarcophagus-org package,
+// the types in this file and onchain-data can get updated
+const curseIsActive = (sarcophagus: any, archaeologist: any): boolean => {
+  return archaeologist.unencryptedShard === '0x' &&
+    [SarcophagusState.Active, SarcophagusState.Resurrecting].includes(sarcophagus.state);
+}
+
+const endOfGracePeriod = (sarcophagus: any, gracePeriod: BigNumber): number => {
+  return sarcophagus.resurrectionTime.toNumber() + gracePeriod.toNumber();
+}
+
+export async function fetchSarcophagiAndScheduleUnwraps(
+  web3Interface: Web3Interface
+): Promise<SarcophagusData[]> {
+  inMemoryStore.gracePeriod = inMemoryStore.gracePeriod || await getGracePeriod(web3Interface);
+
+  const sarcophagi: SarcophagusData[] = [];
+  const sarcoIds = await getSarcophagiIds(web3Interface);
+
+  sarcoIds.map(async sarcoId => {
+    const sarcophagus = await web3Interface.viewStateFacet.getSarcophagus(sarcoId);
+    const archaeologist = await web3Interface.viewStateFacet.getSarcophagusArchaeologist(
+      sarcoId,
+      web3Interface.ethWallet.address
+    );
+
+    if (curseIsActive(sarcophagus, archaeologist)) {
+      const now = new Date().getTime() / 1000;
```

Comment:
This is just a style thing but I find myself often getting confused by the switch between javascript's ms and solidity's seconds. On my branch I've been super explicit about units in my variable names, so I'd call this nowSeconds. It's not really a convention we've agreed on though so your call on whether you think it's helpful or redundant

Reply:
agree! this is updated to `nowSeconds`